src/android/Geolocation.java

```diff
@@ -34,10 +33,13 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 
+
 public class Geolocation extends CordovaPlugin implements OnLocationResultEventListener {
 
     private SparseArray<LocationContext> locationContexts;
-    private FusedLocationProviderClient fusedLocationClient;
+    private FusedLocationProviderClient fusedLocationClientGms;
+
+    private com.huawei.hms.location.FusedLocationProviderClient fusedLocationProviderClientHms;
```

Comment:
maybe we could name the variable `fusedLocationClientHms`? `fusedLocationProviderClientHms` makes it look like a totally different thing from `fusedLocationClientGms` when in reality the only difference is that one is Google's and the other Huawei's.